server/modules/nodeDef/service/nodeDefService.js

```diff
@@ -150,6 +151,16 @@ export const markNodeDefDeleted = async ({ user, surveyId, cycle, nodeDefUuid },
     return afterNodeDefUpdate({ survey: surveyUpdated, nodeDef, nodeDefsDependent, nodeDefsUpdated }, t)
   })
 
+export const markNodeDefsDeleted = async ({ user, surveyId, cycle, nodeDefUuids }, client = db) => {
+  let response = { nodeDefsUpdated: {}, nodeDefsValidation: {} }
+
+  await PromiseUtils.each(nodeDefUuids, async (nodeDefUuid) => {
+    response = await markNodeDefDeleted({ user, surveyId, cycle, nodeDefUuid }, client)
```

Comment:
I would "merge" the response, adding nodeDefsUpdated and nodeDefsValidation to the "accumulator" above